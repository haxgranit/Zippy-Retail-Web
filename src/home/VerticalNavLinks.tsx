import Nav from 'react-bootstrap/Nav';
import { Link, NavLink, useLocation } from 'react-router-dom';
import ZippyCashLogo from '../assets/img/general/ZippyCash_Logo.svg';
import { Version } from '../Version';

export default function VerticalNavLinks() {
  const { pathname } = useLocation();

  return (
    <div className="vertical-nav-wrapper">
      <Nav className="vertical-nav">
        <div className="nav-items noselect">
          <Link to="/my-wallet" className="navbar-brand">
            <img src={ZippyCashLogo} alt="" />
          </Link>
          <Nav.Link className="nav-link" id="my-account-link" as={NavLink} to="/my-account">
            <i className="zippy-cash-icon zc-home" />
            My Account
            <i className="dropdown zippy-cash-icon zc-up" />
          </Nav.Link>
          {pathname.startsWith('/my-account') && (
            <Nav className="flex-column" style={{ marginLeft: '20px' }}>
              <Nav.Link as={NavLink} to="/my-account/account">Account</Nav.Link>
              <Nav.Link as={NavLink} to="/my-account/download-transactions">Download Transactions</Nav.Link>
              <Nav.Link as={NavLink} to="/my-account/view-estatements">View E-Statements</Nav.Link>
            </Nav>
          )}
          <Nav.Link className="nav-link" as={NavLink} to="/my-wallet">
            <i className="zippy-cash-icon zc-interac-etransfer" />
            My Wallet
            <i className="dropdown zippy-cash-icon zc-up" />
          </Nav.Link>
          {pathname.startsWith('/my-wallet') && (
            <Nav className="flex-column" style={{ marginLeft: '20px' }}>
              <Nav.Link as={NavLink} to="/my-wallet/zippy-money">Zippy Money</Nav.Link>
              <Nav.Link as={NavLink} to="/my-wallet/transaction-history">Transaction History</Nav.Link>
              <Nav.Link as={NavLink} to="/my-wallet/load">Load</Nav.Link>
              <Nav.Link as={NavLink} to="/my-wallet/get">Get</Nav.Link>
              <Nav.Link as={NavLink} to="/my-wallet/funding-source">Funding Source</Nav.Link>
              <Nav.Link as={NavLink} to="/my-wallet/auto-deposit">Auto Deposit</Nav.Link>
            </Nav>
          )}
          <Nav.Link className="nav-link" as={NavLink} to="/bill-payments">
            <i className="zippy-cash-icon zc-bill-payments" />
            Bill Payments
          </Nav.Link>
          <Nav.Link className="nav-link" as={NavLink} to="/account-security">
            <i className="zippy-cash-icon zc-account-security" />
            Account Security
          </Nav.Link>
          <hr />
          <Nav.Link className="nav-link" as={NavLink} to="/customer-services">
            <i className="zippy-cash-icon zc-customer-services" />
            Customer Services
          </Nav.Link>
          <Nav.Link className="nav-link" as={NavLink} to="/manage-my-alerts">
            <i className="zippy-cash-icon zc-manage-my-alerts" />
            Manage My Alerts
          </Nav.Link>
          <Nav.Link className="nav-link" as={NavLink} to="/contact-us">
            <i className="zippy-cash-icon zc-contact-us" />
            Contact Us
          </Nav.Link>
        </div>
        <div className="copyright">
          <Version />
          <div className="noselect">
            Â© 2021 Zippy Cash LLC
          </div>
        </div>
      </Nav>
    </div>
  );
}
