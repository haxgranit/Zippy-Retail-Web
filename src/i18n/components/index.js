import LegalEnCAText from "./legal/LegalEnCAText";
import LegalEnUSText from "./legal/LegalEnUSText";
import LegalEsUSText from "./legal/LegalEsUSText";
import LegalFrCAText from "./legal/LegalFrCAText";

export {
    LegalEnCAText,
    LegalEnUSText,
    LegalEsUSText,
    LegalFrCAText
}
