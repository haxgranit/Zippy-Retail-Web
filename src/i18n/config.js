import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import { LegalEnCAText, LegalEnUSText, LegalEsUSText, LegalFrCAText } from './components';

i18n.use(initReactI18next).init({
  fallbackLng: 'en-US',
  lng: 'en-US',
  resources: {
    'en-CA': {
      translations: {
        ...require('./locales/en-CA/translations.json'),
        legal: {
          text: (
            <>
              <LegalEnCAText />
            </>
          )
        }
      }
    },
    'en-US': {
      translations: {
        ...require('./locales/en-US/translations.json'),
        legal: {
          text: (
            <>
              <LegalEnUSText />
            </>
          )
        }
      }
    },
    'es-US': {
      translations: {
        ...require('./locales/es-US/translations.json'),
        legal: {
          text: (
            <>
              <LegalEsUSText />
            </>
          )
        }
      }
    },
    'fr-CA': {
      translations: {
        ...require('./locales/fr-CA/translations.json'),
        legal: {
          text: (
            <>
              <LegalFrCAText />
            </>
          )
        }
      }
    },
  },
  returnObjects: false,
  returnedObjectHandler: (key, value, options) => value,
  ns: ['translations'],
  defaultNS: 'translations'
});

i18n.languages = ['en-CA', 'en-US', 'es-US', 'fr-CA'];

export default i18n;