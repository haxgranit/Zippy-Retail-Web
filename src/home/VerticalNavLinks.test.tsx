import Enzyme, { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import VerticalNavLinks from './VerticalNavLinks';
import '../i18n/config';

// Configure enzyme for react 17
Enzyme.configure({ adapter: new Adapter() });

describe('VerticalNavLinks Component', () => {
  it('click buttons on VerticalNavLinks', () => {
    const wrapper = shallow(
      <BrowserRouter>
        <VerticalNavLinks />
      </BrowserRouter>,
    );

    const component = wrapper.childAt(0).dive();
    component.find('.nav-link').at(0).simulate('click');
    component.find('.nav-link').at(1).simulate('click');
    component.find('.nav-link').at(2).simulate('click');
    component.find('.nav-link').at(3).simulate('click');
  });
});
