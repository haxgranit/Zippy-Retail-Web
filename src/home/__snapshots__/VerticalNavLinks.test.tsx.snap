// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalNavLinks Component matches the snapshot 1`] = `
<div
  className="vertical-nav-wrapper noselect"
>
  <div
    className="vertical-nav nav"
    onKeyDown={[Function]}
  >
    <a
      className="navbar-brand"
      href="/my-wallet"
      onClick={[Function]}
    >
      <img
        alt=""
        src="ZippyCash_Logo.svg"
      />
    </a>
    <a
      className="nav-link nav-link"
      data-rr-ui-event-key={null}
      disabled={false}
      href="/my-account"
      id="my-account-link"
      onClick={[Function]}
    >
      <i
        className="zippy-cash-icon zc-home"
      />
      My Account
      <i
        className="dropdown zippy-cash-icon zc-up"
      />
    </a>
    <a
      className="nav-link nav-link"
      data-rr-ui-event-key={null}
      disabled={false}
      href="/my-wallet"
      onClick={[Function]}
    >
      <i
        className="zippy-cash-icon zc-interac-etransfer"
      />
      My Wallet
      <i
        className="dropdown zippy-cash-icon zc-up"
      />
    </a>
    <a
      className="nav-link nav-link"
      data-rr-ui-event-key={null}
      disabled={false}
      href="/bill-payments"
      onClick={[Function]}
    >
      <i
        className="zippy-cash-icon zc-bill-payments"
      />
      Bill Payments
      <i
        className="dropdown zippy-cash-icon zc-up"
      />
    </a>
    <a
      className="nav-link nav-link"
      data-rr-ui-event-key={null}
      disabled={false}
      href="/account-security"
      onClick={[Function]}
    >
      <i
        className="zippy-cash-icon zc-account-security"
      />
      Account Security
    </a>
    <hr />
    <a
      className="nav-link nav-link"
      data-rr-ui-event-key={null}
      disabled={false}
      href="/customer-services"
      onClick={[Function]}
    >
      <i
        className="zippy-cash-icon zc-customer-services"
      />
      Customer Services
    </a>
    <a
      className="nav-link nav-link"
      data-rr-ui-event-key={null}
      disabled={false}
      href="/manage-my-alerts"
      onClick={[Function]}
    >
      <i
        className="zippy-cash-icon zc-manage-my-alerts"
      />
      Manage My Alerts
    </a>
    <a
      className="nav-link nav-link"
      data-rr-ui-event-key={null}
      disabled={false}
      href="/contact-us"
      onClick={[Function]}
    >
      <i
        className="zippy-cash-icon zc-contact-us"
      />
      Contact Us
    </a>
    <div
      className="copyright"
    >
      Â© 2021 Zippy Cash LLC
    </div>
  </div>
</div>
`;
